File: articles/what-is-algorithms-computational-reasoning/c/algorithmic_reasoning.c
#include <stdio.h>

typedef struct {
  const char* name;
  double representation;
  double correctness;
  double governance;
  double brute_force;
} Scenario;

static double clamp(double v) {
  if (v < 0.0) return 0.0;
  if (v > 100.0) return 100.0;
  return v;
}

int main(void) {
  Scenario scenarios[] = {
    {"Brute-force procedure", 40, 28, 20, 92},
    {"Indexed search design", 62, 52, 38, 42},
    {"Graph-aware reasoning", 76, 68, 54, 30},
    {"Governed computational reasoning", 86, 82, 86, 18}
  };
  printf("scenario,reasoning_score,warning\n");
  for (int i = 0; i < 4; i++) {
    double score = clamp(0.30 * scenarios[i].representation + 0.30 * scenarios[i].correctness + 0.30 * scenarios[i].governance - 0.10 * scenarios[i].brute_force);
    printf("%s,%.3f,Synthetic governance diagnostic only.\n", scenarios[i].name, score);
  }
  return 0;
}

File: articles/what-is-algorithms-computational-reasoning/typescript/src/algorithmicReasoning.ts
export type Scenario = {
  name: string;
  representation: number;
  correctness: number;
  governance: number;
  bruteForce: number;
};

export function reasoningScore(scenario: Scenario): number {
  const score =
    0.30 * scenario.representation +
    0.30 * scenario.correctness +
    0.30 * scenario.governance -
    0.10 * scenario.bruteForce;
  return Math.max(0, Math.min(100, score));
}

export const scenarios: Scenario[] = [
  { name: "Brute-force procedure", representation: 40, correctness: 28, governance: 20, bruteForce: 92 },
  { name: "Indexed search design", representation: 62, correctness: 52, governance: 38, bruteForce: 42 },
  { name: "Graph-aware reasoning", representation: 76, correctness: 68, governance: 54, bruteForce: 30 },
  { name: "Governed computational reasoning", representation: 86, correctness: 82, governance: 86, bruteForce: 18 }
];
